controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -223,14 +223,13 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 		profileName = r.storageCluster.Spec.DefaultStorageProfile
 	}
 
-	for i := range r.storageCluster.Spec.StorageProfiles {
-		profile := &r.storageCluster.Spec.StorageProfiles[i]
-		if profile.Name == profileName {
-			r.storageProfile = profile
-			break
-		}
+	storageProfileList := &v1.StorageProfileList{}
+	if err := r.list(storageProfileList, client.InNamespace(r.storageCluster.GetNamespace())); err != nil {
+		return reconcile.Result{}, err
 	}
 
+	r.storageProfile = storageProfileList.Items[profileName]
```

Comment:
As we discussed in an offline review. The reconciler object state is preserved across calls. As such it does not make sense to save anything on the reconciler that is specific to a single CR instance as it can negatively affect the reconciliation of the next event (for a different `StorageClassRequest` instance)

There are 3 approaches that can be used here instead:
1. Load all of the Profiles into a map with the Metadata.Name as key 
2. Load only the named profile (use `r.get` instead of `r.list`) as a variable and  just pass it into the reconcile functions as parameters
3. Load only the named profile and add it to the context object using `ctx.WithValue` 

For this use case, my recommendation is to go with option 2, it is very straight forward, very readable and because all of out reconcile functions are called from reconcile phases it will not add complexity to the codebase